rustls-mbedcrypto-provider/src/fips_utils/mod.rs

```diff
@@ -318,18 +322,18 @@ mod tests {
         assert_eq!("Other(OtherError(Mbedtls(AesBadInputData)))", rustls_test_dbg);
     }
 
-    fn create_ffdhe_key_pair(dhe_group: &DheKxGroup) -> (Mpi, Mpi) {
-        let g = Mpi::from_binary(dhe_group.group.g).unwrap();
-        let p = Mpi::from_binary(dhe_group.group.p).unwrap();
+    fn create_ffdhe_key_pair<T: RngCallback>(dhe_group: &DheKxGroupWrapper<T>) -> (Mpi, Mpi) {
+        let g = Mpi::from_binary(dhe_group.dhe_kx_group.group.g).unwrap();
+        let p = Mpi::from_binary(dhe_group.dhe_kx_group.group.p).unwrap();
         let mut rng = crate::rng::rng_new().unwrap();
-        let mut x_binary = vec![0; dhe_group.priv_key_len];
+        let mut x_binary = vec![0; dhe_group.dhe_kx_group.priv_key_len];
         rng.random(&mut x_binary).unwrap();
         print_vec("private", &x_binary);
```

Comment:
This is a test function but please remove these `print_vec` statements with secret values